features/crashprobe.feature

```diff
@@ -100,99 +90,97 @@ Scenario: Stack overflow
+    When I run "ObjCMsgSendScenario" and relaunch the app
+    And I configure Bugsnag for "ObjCMsgSendScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
     And the exception "errorClass" equals "EXC_BAD_ACCESS"
-    And the exception "message" equals "Attempted to dereference garbage pointer 0x42."
+    And the exception "message" equals "Attempted to dereference garbage pointer 0x38."
     And the "method" of stack frame 0 equals "objc_msgSend"
 
 Scenario: Attempt to execute an instruction undefined on the current architecture
-    When I crash the app using "UndefinedInstructionScenario"
-    And I relaunch the app
-    And I wait for a request
-    Then the request is valid for the error reporting API
+    When I run "UndefinedInstructionScenario" and relaunch the app
+    And I configure Bugsnag for "UndefinedInstructionScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
     And the exception "errorClass" equals "EXC_BAD_INSTRUCTION"
     And the "method" of stack frame 0 equals "-[UndefinedInstructionScenario run]"
 
 Scenario: Send a message to an object whose memory has already been freed
-    When I crash the app using "ReleasedObjectScenario"
-    And I relaunch the app
-    And I wait for a request
-    Then the request is valid for the error reporting API
+    When I run "ReleasedObjectScenario" and relaunch the app
+    And I configure Bugsnag for "ReleasedObjectScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
     And the exception "message" starts with "Attempted to dereference garbage pointer"
     And the exception "errorClass" equals "EXC_BAD_ACCESS"
     And the "method" of stack frame 0 equals "objc_msgSend"
-    And the "method" of stack frame 1 equals "-[ReleasedObjectScenario run]"
+    And the "method" of stack frame 1 equals "__29-[ReleasedObjectScenario run]_block_invoke"
 
 # N.B. this scenario is "imprecise" on CrashProbe due to line number info,
 # which is not tested here as this would require symbolication
 Scenario: Crash within Swift code
-    When I crash the app using "SwiftCrash"
-    And I relaunch the app
-    And I wait for a request
-    Then the request is valid for the error reporting API
-    And the exception "message" contains "Unexpectedly found nil while unwrapping an Optional value"
+    When I run "SwiftCrash" and relaunch the app
+    And I configure Bugsnag for "SwiftCrash"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
+    # And the exception "message" equals "Unexpectedly found nil while unwrapping an Optional value"
```

Comment:
Various commented lines in this file - should be reinstated or removed if appropriate (but I'm guessing they currently fail?).